_overviews/scala3-macros/best-practices.md

```diff
@@ -55,7 +55,100 @@ val y: Expr[Int] = ...
 '{ (x: Int) => ${ body('x, y) } }
 ```
 
+## Quotes Reflect
 
+For this section, consider the following setup:
 
-## Quotes Reflect
-**Coming soon**
+```scala
+object Box:
+  sealed trait Base
+  case class Leaf(x: Int) extends Base
+
+// Quotes in contextual scope
+val boxTpe = TypeRepr.of[Box.type]
+val baseTpe = TypeRepr.of[Box.Base]
+val baseSym = baseTpe.typeSymbol
+val leafTpe = TypeRepr.of[Box.Leaf]
+val leafSym = leafTpe.typeSymbol
+```
+
+### Avoid `Symbol.tree`
+
+`Symbol.tree` returns the `Tree` associated to the symbol. Be careful when using this
+method as the tree for a symbol might not be defined. When the code associated to the symbol
+is defined in a different moment than this access, if the `-Yretain-trees` compilation option
+is not used, then the `tree` of the symbol will not be available. Symbols originated from
+Java code do not have an associated `tree`.
+
+### Obtaining a `TypeRepr` from a `Symbol`
+
+In the previous paragraph we saw that `Symbol.tree` should be avoided and therefore
+you should not use `Symbol.tree.tpe`.
+Thus to obtain the `TypeRepr` corresponding to a `Symbol`, it is recommended
+to use `TypeRepr.memberType`
+
+We can obtain the `TypeRepr` of `Leaf` in two ways:
```

Comment:
Is there a difference between both ways?

Reply:
I meant to say that the two ways are equivalent. The first option is available only if you that there is `Box.Leaf` while the second one allows to explore the api only when you have only a symbol

Reply:
I see! I suggest to mention this difference explicitly in the text, even though it is obvious when looking at the type signatures.